src/js/index.js

```diff
@@ -1,36 +1,69 @@
-import { RequestManager, HTTPTransport, Client } from "@open-rpc/client-js";
-
-const transport = new HTTPTransport(process.env.FAUCET_NODE_URL);
-const client = new Client(new RequestManager([transport]));
-
+import {formatDuration, intervalToDuration} from "date-fns";
 
 const requestForm = document.getElementById('request-form');
 const successMessage = document.getElementById('success-message');
 const errorMessage = document.getElementById('error-message');
+const rateLimitedErrorMessage = document.getElementById('rate-limit-error-message');
+const rateLimitTime = document.getElementById('rate-limit-time');
 const loader = document.getElementsByClassName('loader')[0];
 const messageAddress = document.getElementById('message-address');
 const messageTxId = document.getElementById('message-txid');
 const submitButton = document.getElementById('form-submit');
 
-requestForm.onsubmit = (event) => {
+const formatBackoffTime = (seconds) => formatDuration(intervalToDuration({ start: 0, end: seconds * 1000 }))
+
+requestForm.onsubmit = async (event) => {
   event.preventDefault();
   successMessage.style.display = 'none';
   errorMessage.style.display = 'none';
+  rateLimitedErrorMessage.display = 'none';
   loader.style.display = 'block';
   submitButton.disabled = true;
   const address = document.getElementById('address-field').value
-  client.request({method: 'faucet_sendFunds', params: [address]})
-    .then((result) => {
-      messageAddress.textContent = `Address: ${address}`
-      messageTxId.textContent = `Transaction ID: ${result}`
-      successMessage.style.display = 'block';
-    })
-    .catch((err) => {
+  const requestBody = {
+   	"jsonrpc": "2.0",
+    "method": "faucet_sendFunds", 
+    "params": [address],
+    "id": 1
```

Comment:
Are you using prettier?

Reply:
I will add it :) The goal was to have as minimal project as possible here, but it's just tedious to format all the code manually :)